.github/workflows/cd.yml

```diff
@@ -34,40 +34,87 @@ jobs:
+          sudo dpkg -x libdbus-1-3*.deb /build/deps_root/
+          sudo dpkg -x libdbus-1-dev*.deb /build/deps_root/
+          sudo dpkg -x libsystemd0*.deb /build/deps_root/
+          sudo dpkg -x libgcrypt20_*.deb /build/deps_root/
+          sudo dpkg -x liblzma5_*.deb /build/deps_root/
+          sudo dpkg -x liblz4-1_*.deb /build/deps_root/
+          sudo dpkg -x libgpg-error0_*.deb /build/deps_root/
+          sudo rm /build/deps_root/usr/lib/arm-linux-gnueabihf/libdbus-1.so
+          sudo ln -rs /build/deps_root/lib/arm-linux-gnueabihf/libdbus-1.so.3 /build/deps_root/usr/lib/arm-linux-gnueabihf/libdbus-1.so
+          sudo ln -rs /build/deps_root/lib/arm-linux-gnueabihf/libgcrypt.so.20 /build/deps_root/lib/arm-linux-gnueabihf/libgcrypt.so
+          sudo ln -rs /build/deps_root/lib/arm-linux-gnueabihf/libgpg-error.so.0 /build/deps_root/lib/arm-linux-gnueabihf/libgpg-error.so
+          sudo ln -rs /build/deps_root/lib/arm-linux-gnueabihf/liblzma.so.5 /build/deps_root/lib/arm-linux-gnueabihf/liblzma.so
+          sudo ln -rs /build/deps_root/lib/arm-linux-gnueabihf/libsystemd.so.0 /build/deps_root/lib/arm-linux-gnueabihf/libsystemd.so
+          sudo ln -rs /build/deps_root/usr/lib/arm-linux-gnueabihf/liblz4.so.1 /build/deps_root/usr/lib/arm-linux-gnueabihf/liblz4.so
+          mkdir -p .cargo
+          echo -e '[target.arm-unknown-linux-gnueabihf.dbus]\nrustc-link-lib = ["dbus-1", "gcrypt", "gpg-error", "lz4", "lzma", "systemd"]' >> .cargo/config
       - name: Checking out sources
         uses: actions/checkout@v1
       - name: Running cargo build
+        if: matrix.target != 'arm-unknown-linux-gnueabihf'
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          toolchain: ${{ matrix.rust }}
+          args: --locked --release --target ${{ matrix.target }} --no-default-features --features "${{ matrix.feature }},${{ matrix.audio_backend }}_backend"
+      - name: Running cargo cross build
+        if: matrix.target == 'arm-unknown-linux-gnueabihf'
         uses: actions-rs/cargo@v1
         with:
           command: build
           toolchain: ${{ matrix.rust }}
           args: --locked --release --target ${{ matrix.target }} --no-default-features --features "${{ matrix.feature }},${{ matrix.audio_backend }}_backend"
-      
+        env:
+          RUSTFLAGS: "-C linker=arm-linux-gnueabihf-gcc -L/usr/arm-linux-gnueabihf/lib -L/build/deps_root/usr/lib/arm-linux-gnueabihf -L/build/deps_root/lib/arm-linux-gnueabihf"
+          PKG_CONFIG_ALLOW_CROSS: 1
+          OPENSSL_LIB_DIR: /build/deps_root/usr/lib/arm-linux-gnueabihf
+          OPENSSL_INCLUDE_DIR: /build/deps_root/usr/include/arm-linux-gnueabihf
       - name: Packaging final binary
         shell: bash
         run: |
           cd target/${{ matrix.target }}/release
-          
-          strip spotifyd
```

Comment:
`strip` will not recognise cross-compiled binaries, so we cannot always use it (and just assume that Rust creates good-enough ones).